corgi/core/models.py

```diff
@@ -347,70 +347,40 @@ def get_related_components(build_ids: QuerySet, sort_field: str) -> QuerySet["Co
         # Else self.builds is an empty QuerySet
         return Component.objects.none()
 
-    @staticmethod
-    def get_product_component_relations(
-        variant_ids: list[str], sort_field="external_system_id", only_errata=False
-    ) -> QuerySet["ProductComponentRelation"]:
-        """Returns unique productcomponentrelations with at least 1 matching variant_id,
-        ordered by sort_field. if only_errata is True, only return PCRs with Type=ERRATA
-        """
-        if variant_ids:
-            query = Q()
-            for variant_id in variant_ids:
-                query |= Q(product_ref__contains=variant_id)
-            result = ProductComponentRelation.objects.filter(query).distinct().order_by(sort_field)
-            if only_errata:
-                result = result.filter(type=ProductComponentRelation.Type.ERRATA)
-            return result
-        # Else self.product_variants is an empty QuerySet
-        return ProductComponentRelation.objects.none()
-
     @property
     def builds(self) -> QuerySet:
-        """Return unique build IDs for errata with variant_ids matching self.product_variants"""
-        return (
-            self.get_product_component_relations(
-                self.product_variants, "build_id", only_errata=True
+        """Returns unique productcomponentrelations with at least 1 matching variant or stream,
+        ordered by build_id.
+        """
+        product_refs = [self.name]
+        if isinstance(self, ProductStream):
+            # we also want to include child product_variants for each product stream in the query
+            product_refs.extend(self.product_variants)
+        elif not isinstance(self, ProductVariant):
+            # If it's a product variant the product_variants list will be None
+            product_refs = self.product_streams + self.product_variants
+
```

Comment:
Code above looks almost right, but we don't look up by self.name when self is a ProductVersion or Product. I can't tell if that's a typo or not.

Algorithm for what you have above:
1. Start with only self.name in the lookup.
2. If it's a stream, just add the child variants to the lookup.
3. Else if it's not a stream and not a variant (so must be either a product or product version), add both the child streams and child variants to the lookup. TODO: You overwrite ```product_refs =``` instead of appending ```product_refs +=```, so we lose self.name in this case. Is that a bug?
5. Else it must have been a variant, and none of the if-blocks will be entered so the lookup will only contain the variant's name.

Assuming 3. was on purpose, below might be a bit easier to follow since it avoids the negative if case and documents the else case with a comment:
```suggestion
        if isinstance(self, ProductStream):
            # we also want to include child product_variants for each product stream in the query
            product_refs.extend(self.product_variants)
        elif isinstance(self, ProductVersion) or isinstance(self, Product):
            # we also want to include child product_streams and product_variants in the query
            product_refs = self.product_streams + self.product_variants
        # else it was a product variant, only look up by self.name
```

Or if 3. was a bug, just add the += on the second-to-last line.

Reply:
good idea, I'll use this.